liblangutil/Scanner.cpp

```diff
@@ -271,12 +273,67 @@ bool Scanner::skipWhitespaceExceptUnicodeLinebreak()
 	return sourcePos() != startPosition;
 }
 
+struct Scanner::UnicodeDirectionOverrideProcessor
+{
+	Scanner& scanner;
+	int rtlOverrideDepth = 0;
+
+	pair<bool, ScannerError> operator()()
+	{
+		// U+202D (LRO - Left-to-Right Override)
+		// U+202E (RLO - Right-to-Left Override)
+		if (
+			scanner.tryScanByteSequence("\xE2\x80\xAD") ||
+			scanner.tryScanByteSequence("\xE2\x80\xAE")
+		)
+		{
+			rtlOverrideDepth++;
+			return pair{true, ScannerError::NoError};
+		}
+		else if (scanner.tryScanByteSequence("\xE2\x80\xAC")) // U+202C (PDF - Pop Directional Formatting)
+		{
+			rtlOverrideDepth--;
+			if (rtlOverrideDepth < 0)
+				return pair{true, ScannerError::DirectionalOverrideUnderflowInComment};
+			else
+				return pair{true, ScannerError::NoError};
+		}
+		else
+			return pair{false, ScannerError::NoError};
+	}
+
+	ScannerError error() const noexcept
+	{
+		if (rtlOverrideDepth < 0)
+			return ScannerError::DirectionalOverrideUnderflowInComment;
+		else if (rtlOverrideDepth > 0)
+			return ScannerError::DirectionalOverrideMismatchInComment;
+		else
+			return ScannerError::NoError;
+	}
+};
+
 Token Scanner::skipSingleLineComment()
 {
+	UnicodeDirectionOverrideProcessor unicodeDirectionalOverride{*this};
+
 	// Line terminator is not part of the comment. If it is a
 	// non-ascii line terminator, it will result in a parser error.
 	while (!isUnicodeLinebreak())
-		if (!advance()) break;
+	{
+		auto const [processed, errorCode] = unicodeDirectionalOverride();
+		if (processed && errorCode != ScannerError::NoError)
+			return setError(errorCode);
+		else if (!processed)
+		{
+			if (!advance())
+				break;
+		}
```

Comment:
```suggestion
		else if (!processed && !advance())
			break;		
```

Reply:
I intentionally decided against that, because advance() contains a side effect that I think isn't that obvious hidden into a non-trivial expression.

Reply:
In that case, maybe just get rid of the extra `{` `}`?

Reply:
I think this is subjective. I don't want to argue here, because I intentionally sometimes use `{}`'s to get the body visually more distinct from the if-condition, but yeah ... :-)